### ISSUE TITLE
TreeTab clients contains clients from other (all?) groups

### ISSUE THREAD

**Issue by pawamoy** [qtile/qtile#1459]
Noticed it while automating screenshots.

I rely on the result of `client.layout.info().get("clients")` to know if I should continue killing window with `client.window.kill()` (to clean up the group).

When the TreeTab layout is active, the "clients" contains clients from other, if not all groups.

Example to reproduce:
- open two windows with shells in group a, run `sleep 1000` in one and `sleep 2000` in the other
- open one window with a shell in group d, run `sleep 3000`
- go to group s, *pick treetab layout*, open two windows, one with `sleep 4000`, and run `qtile-cmd -o layout -f info` in the other one

I get this output:

```
{'client': ['pawamoy@zsh - /home/pawamoy/data/dev/forks/qtile: sleep 4000',
            'pawamoy@zsh - /home/pawamoy/data/dev/forks/qtile: sleep 5000',
            'pawamoy@zsh - /home/pawamoy/data/dev/forks/qtile: qtile-cmd -o layout -f info',
            'pawamoy@zsh - /home/pawamoy/data/dev/forks/qtile: sleep 1000',
            'pawamoy@zsh - /home/pawamoy/data/dev/forks/qtile: sleep 2000',
            'pawamoy@zsh - /home/pawamoy/data/dev/forks/qtile: sleep 3000'],
 'group': 's',
 'name': 'treetab',
 'sections': ['Default']}
```

The order of actions has no importance. You can open windows before/after going to group s or switching to treetab layout.

**Comment by pawamoy** [comment]
Same issue for floating layout.

**Comment by whatacold** [comment]
I'm a bit confused by TreeTab's code, it seems to me that they are organized just as lists for each sections, not trees as implied by the name.

Do I misunderstand it?

**Comment by whatacold** [comment]
> ... it seems to me that they are organized just as lists for each sections, not trees as implied by the name.

Indeed, it's a tree structure as commands `cmd_move_left`/`cmd_move_right` can move the current window up/down a level of the tree.(Well, it also can be seen as a forest as there can be more than one sections.)

**Comment by pawamoy** [comment]
This should still be fixed.
